frontend/js/src/user/components/AddSingleListen.tsx

```diff
@@ -47,6 +52,13 @@ export default function AddSingleListen({
     };
   }, [initialText]);
 
+  useImperativeHandle(ref, () => ({
+    reset: () => {
```

Comment:
Along with my comment above, I think this is where you'll want to call the new function to reset the search component state
(`searchInputRef?.current?.reset()`)